dev/RatingControl/RatingControl.cpp

```diff
@@ -46,6 +47,34 @@ float RatingControl::ActualRatingFontSize()
     return RenderingRatingFontSize() / 2;
 }
 
+double RatingControl::ItemSpacing()
+{
+    // TextScaleFactor will change font size but won't affect spacing/margin.
+    // The scaled-up glyph will grow out to the margin area, so the "actual" margins become smaller.
+    // Therefore we should include TextScaleFactor when calculating item spacing in order to get correct total width and star center positions.
+    return c_defaultItemSpacing / GetUISettings().TextScaleFactor();
+}
+
+void RatingControl::UpdateCaptionMargins()
+{
+    // star glyph's vertical center position is set to 0.8 per redline, but caption text's center is 0.5 (default value).
+    // when they scale up the stars and caption text won't be vertically aligned since the vertical center is different.
+    // Update caption text top margin here to fix the alignment.
+    if (auto captionTextBlock = m_captionTextBlock.safe_get())
+    {
+        double textScaleFactor = GetUISettings().TextScaleFactor();
+        winrt::Thickness margin = captionTextBlock.Margin();
+        margin.Top = c_defaultCaptionTopMargin;
+
+        if (textScaleFactor > 1.01)
```

Comment:
>1.01 [](start = 30, length = 4)

1.01 seems like a weird number, why isn't this just > 1.0 ?

Reply:
The actual 1.0 value is something like 1.00000001, I tried > 1.0 that didn't really work very well, so changed it to 1.01 :\ 

I'm trying to update the function to make it work with all scales like Jevan said, so this special value check will be gone.

Reply:
If you need to use this I think using Double.Epsilon would be better. something like if(textScaleFactor > 1.0 + Double.Epsilon)

---
In reply to: [248482045](https://github.com/Microsoft/microsoft-ui-xaml/pull/187#discussion_r248482045) [](ancestors = 248482045)